st2auth/st2auth/backends/__init__.py

```diff
@@ -60,10 +66,22 @@ def get_backend_instance(name):
         try:
             kwargs = json.loads(backend_kwargs)
         except ValueError as e:
-            raise ValueError('Failed to JSON parse backend settings: %s' % (str(e)))
+            raise ValueError('Failed to JSON parse backend settings for backend "%s": %s' %
+                             (name, str(e)))
     else:
         kwargs = {}
 
     cls = manager.driver
-    cls_instance = cls(**kwargs)
+
+    try:
+        cls_instance = cls(**kwargs)
+    except Exception as e:
+        tb_msg = traceback.format_exc()
+        class_name = cls.__name__
+        msg = ('Failed to instantiate auth backend "%s" (class %s) with backend settings '
+               '"%s": %s' % (name, class_name, str(kwargs), str(e)))
+        msg += '\n\n' + tb_msg
+        exc_cls = type(e)
+        raise exc_cls(msg)
+
     return cls_instance
```

Comment:
I realize we can't reach this because we raise in the exception, but it feels weird having it outside the `try` block. Would be more explicit to have it there and if for whatever reason the raise ever gets removed it won't try to return an unset variable.